### PR TITLE
GEODE-6804: add tab completion for deploy --jar= and --dir=

### PR DESCRIPTION
**Pull request by onichols-pivotal**
Before, no tab completion was available, in contrast to other commands like `start server --properties-file=`

After, filesystem-based completion is available, with the following rules:
- `--jars` supports a comma-separated list of .jar files, although you must type the comma to start the next completion cycle
- `--dir` will avoid suggesting directories that don't contain any .jar files

During tab completion of either jars or dirs, it will look ahead one directory level to see if it can filter out directories that contain neither jars nor further subdirectories.

### CODE REVIEW

**Comment by onichols-pivotal** [comment]
@jmelchio

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarDirPathConverter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.shell.core.Completion;
+import org.springframework.shell.core.Converter;
+import org.springframework.shell.core.MethodTarget;
+
+import org.apache.geode.management.cli.ConverterHint;
+
+public class JarDirPathConverter implements Converter<String> {
```
Comment (jmelchio) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarDirPathConverter.java]: Unit test missing for Class.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarFilesPathConverter.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.shell.core.Completion;
+import org.springframework.shell.core.Converter;
+import org.springframework.shell.core.MethodTarget;
+
+import org.apache.geode.management.cli.ConverterHint;
+
+public class JarFilesPathConverter implements Converter<String[]> {
```
Comment (jmelchio) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarFilesPathConverter.java]: Unit test missing for Class.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarDirPathConverter.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarDirPathConverter.java]: unit test added

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarFilesPathConverter.java (hunk 2 above)
Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/converters/JarFilesPathConverter.java]: unit test added

**Pull request merged**